mycroft/skills/mycroft_skill/mycroft_skill.py

```diff
@@ -585,9 +588,14 @@ def voc_match(self, utt, voc_filename, lang=None):
             vocab = read_vocab_file(voc)
             self.voc_match_cache[cache_key] = list(chain(*vocab))
         if utt:
-            # Check for matches against complete words
-            return any([re.match(r'.*\b' + i + r'\b.*', utt)
-                        for i in self.voc_match_cache[cache_key]])
+            if exact:
+                # Check for exact match
+                return utt and any(i.strip() == utt
```

Comment:
Do we need the `utt and` given line 590 checks for the existence of `utt` already?

Reply:
good catch